tests/shell/shell_recorder/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+if (ENABLE_KDB_TESTING)
+        # set kdb command
+        SET(USE_CMAKE_KDB_COMMAND "")
+        if (BUILD_FULL)
+                set (KDB_COMMAND "${CMAKE_BINARY_DIR}/bin/kdb-full")
+        elseif (BUILD_STATIC)
+                set (KDB_COMMAND "${CMAKE_BINARY_DIR}/bin/kdb-static")
+        else (BUILD_SHARED)
+                set (KDB_COMMAND "${CMAKE_BINARY_DIR}/bin/kdb")
+        else()
+                message(SEND_ERROR "no kdb tool found, please enable BUILD_FULL, BUILD_STATIC or BUILD_SHARED")
+        endif ()
+        configure_file (
+                        "${CMAKE_CURRENT_SOURCE_DIR}/shell_recorder.sh"
+                        "${CMAKE_CURRENT_BINARY_DIR}/shell_recorder.sh"
+                        @ONLY
+                )
+
+        file (GLOB SCRIPT_TESTS *.dat)
```

Comment:
to me .dat is like foo or bar, i just needed a fileending and didn't want to think about it. .esr seems fine to me.